portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,95 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var context;
+var canvas;
+var numBubs = 20;
+var interval_ID = null;
+// Add a event listener to start from top when window is resized
+window.addEventListener('resize', startChat)
+
+function startChat() {
+  grabComments();
+  canvas = document.getElementById('myCanvas');
+  context = canvas.getContext('2d');
+  canvas.width = window.innerWidth;
+  canvas.height = window.innerHeight;
+  init();
+}
+
+class Bubble {
+  constructor(color) {
+    this.speed_factor = 3;
+    if (canvas.width < 900) {
+      this.speed_factor = 1.75;
+    }
+    this.color = color;
+    this.size = Math.random() * canvas.width;
+    this.x = Math.random() * canvas.width;
+    this.y = Math.random() * canvas.height;
+    this.radius = Math.floor(Math.random() * Math.sqrt(canvas.width));
+    this.dx = Math.random() * this.speed_factor * this.randomDirection();
+    this.dy = Math.random() * this.speed_factor * this.randomDirection();
+    this.opacity = Math.random();
+  }
+  drawBubble() {
+    context.beginPath();
+    context.fillStyle = this.color;
+    context.globalAlpha = this.opacity;
+    context.arc(this.x, this.y, this.radius, 0, Math.PI * 2, true);
+    context.closePath();
+    context.fill();
+  }
+  randomDirection() {
+    if (Math.random() > .5) {
+      return -1;
+    } else {
+      return 1;
+    }
+  }
+}
+
+function init() {
+  if (interval_ID !== null) {
+    clearInterval(interval_ID);
+  }
+  var myBubbleArray = [];
+  for (var i = 0; i < numBubs; i++) {
+    myblue = new Bubble('#4885ed');
+    myred = new Bubble('#db3236');
+    mygreen = new Bubble('#3cba54');
+    myyellow = new Bubble('#f4c20d');
+    myBubbleArray.push(myblue);
+    myBubbleArray.push(myred);
+    myBubbleArray.push(mygreen);
+    myBubbleArray.push(myyellow);
+  }
+  interval_ID = setInterval(move, 20, myBubbleArray);
```

Comment:
Not a big deal, but setInterval every 20 milliseconds could cause performance issues on lower end devices. It can also make the bubbles move more/less because move is not called every 20ms exactly.

Better practice would be to use [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) to repeatedly enqueue calls to `move`, and use the time elapsed since the last call to